### ISSUE TITLE
Kafka source/sink TLS configuration regression

### ISSUE THREAD

**Issue by fpytloun** [vectordotdev/vector#14238]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

Previously there was `tls.enabled = true` option for sinks which disappeared (and I didn' notice in breaking changes), however when removing that and keeping just:
```
      tls.ca_file = "/secrets/identity/server_ca_with_compass.crt"
      tls.crt_file = "/secrets/identity/client.crt"
      tls.key_file = "/secrets/identity/client.key"
```
doesn't pass SSL options to rdkafka:
```
2022-08-31T14:44:47.174121Z ERROR rdkafka::client: librdkafka: Global error: BrokerTransportFailure (Local: Broker transport failure): kafka-1:9093/bootstrap: Disconnected while requesting ApiVersion: might be caused by incorrect security.protocol configuration (connecting to a SSL listener?) or broker version is < 0.10 (see api.version.request) (after 78ms in state APIVERSION_QUERY)
```

Can be workarounded by setting rdkafka options directly:
```
      librdkafka_options."security.protocol" = "ssl"
      librdkafka_options."ssl.ca.location" = "/secrets/identity/server_ca_with_compass.crt"
      librdkafka_options."ssl.certificate.location" = "/secrets/identity/client.crt"
      librdkafka_options."ssl.key.location" = "/secrets/identity/client.key"
```

### Version

0.24.0

**Comment by jszwedko** [comment]
Hi @fpytloun !

Actually the `kafka` source and sink do support `tls.enabled`. That was an incorrect adjustment to the docs. I'll replace it.

Does it correctly pass the options if you provide `tls.enabled = true`?

**Comment by fpytloun** [comment]
@jszwedko yes, it seems that `tls.enabled` is still present. I initially got error that this option is unknown (but maybe from different sink, not sure) so I assumed according to docs that it was removed from all sinks.

**Comment by jszwedko** [comment]
👍 Yeah, I fixed the docs in https://github.com/vectordotdev/vector/pull/14242
